[Fixes #40] Adjust stuff for CI / pre-release

michalfita opened the pull request:
Preparing stuff for HAL 0.4.0 release.

michalfita committed:
[Fixes #40] Adjust stuff for CI

michalfita committed:
[Fixes #40] Amend `CHANGELOG.md`

tmplt commented on CHANGELOG.md:
```diff
@@ -2,14 +2,18 @@
 
 - [Sealed](https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed) all HAL-internal traits to disallow potentially unsound downstream trait implementations.
 - Regenerated PACs with `svd2rust 0.25.1` using up-to-date (2022-05-18) SVDs. These SVDs are now vendored in this repository, instead of using a submodule.
+- Cargo Workspace sorted, so local builds and crate publictions work, and CI is satisfied.
```
GitHub CI was added with in this release, so there is no need to mention that a fix was applied.

martinmortsell reviewed (approved):
Looks good to me.

michalfita committed:
[Fixes #40] Cleaning-up

michalfita committed:
[Fixes #34] After PAC 0.25.0 some leftovers required addressing.

This would form the next release, so we can go with next releases with a clean slate.

tmplt commented on CHANGELOG.md:
```diff
@@ -13,7 +13,8 @@
 - [Sealed](https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed) all HAL-internal traits to disallow potentially unsound downstream trait implementations.
 - Regenerated PACs with `svd2rust 0.25.1` using up-to-date (2022-05-18) SVDs. These SVDs are now vendored in this repository, instead of using a submodule.
-- Cargo Workspace sorted, so local builds and crate publictions work, and CI is satisfied.
+- Cargo Workspace sorted, so local builds and crate publictions work.
```
spelling

tmplt commented on tools/update-pacs.sh:
```diff
@@ -0,0 +1,69 @@
+    pushd $pac
+
+    #sed -e "s|\${crate}|${crate}|g" -e "s|\${chip}|${chip}|g" -e "s|\${pac}|${pac}|g" -e "s|\${version}|${version}|g" "${top}/pac/templates/Cargo.toml.template" > Cargo.toml
+    #sed -e "s|\${crate}|${crate}|g" -e "s|\${chip}|${chip}|g" -e "s|\${pac}|${pac}|g" -e "s|\${version}|${version}|g" "${top}/pac/templates/README.md.template" > README.md
```
remove outcommented code

michalfita committed:
[Fixes #34] Address review comments

michalfita commented:
[0.4.0 Released](https://github.com/atsams-rs/atsamx7x-rust/releases/tag/v0.4.0)!
